src/sync_manager.hpp

```diff
@@ -43,7 +44,14 @@ enum class SyncSessionStopPolicy {
     AfterChangesUploaded,           // Once all Realms/Sessions go out of scope, wait for uploads to complete and stop.
 };
 
-using SyncLoginFunction = std::function<void(const std::string&, const SyncConfig&)>;
+// A global callback used to indicate that a network request should be made to bind a session.
+// The callback is passed the current user token of the session's user, the unresolved URL of the
+// session to be bound, a boolean indicating whether this is the initial bind (true) or a refresh,
+// and a shared pointer to a SyncSession object upon which the bind or refresh API should be invoked
+// once the network response is parsed.
+// NOTE: the bind or refresh APIs must be called asynchronously.
+// FIXME: This can go away entirely once the new token system is implemented.
+using SyncBindSessionFunction = std::function<void(const std::string&, const std::string&, bool, std::shared_ptr<SyncSession>)>;
```

Comment:
Passing a `shared_ptr<SyncSession>` to this function is problematic. We currently call the log-in callback from within `SyncSession`'s constructor, but no `shared_ptr` to the object exists at that point.